src/main/java/Task.java

```diff
@@ -0,0 +1,23 @@
+public class Task {
+    protected String taskName;
+    protected boolean isDone;
+
+    public Task(String taskName) {
+        this.taskName = taskName;
+        this.isDone = false;
+    }
+
+    public String getTaskName() {
+        return taskName;
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "X" : " ");
+    }
+
+    public void markAsDone(boolean done) {
+        this.isDone = done;
+    }
+
```

Comment:
Good implementation of the base class. Could further include toString to avoid repetition within sub classes